FuelLabs/fuel-indexer
# forc index check: switch to async reqwest

### Description

Fixes:
```
cargo run -p forc-index -- check
```
```
thread 'main' panicked at 'Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.', /Users/mw/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.1/src/runtime/blocking/shutdown.rs:51:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Testing steps

1. Checkout `master`, run `cargo run -p forc-index -- check`, or `./target/debug/forc-index check`. Both should produce the error.
2. Run the same commands on this branch; there should be no error.

### Changelog

* Use `async` version of `reqwest`.
